hydrus/data/crud.py

```diff
@@ -514,17 +515,22 @@ def update(id_: str,
 def get_collection(API_NAME: str,
                    type_: str,
                    session: scoped_session,
-                   path: str = None) -> Dict[str,
-                                             Any]:
+                   paginate: bool,
+                   page_size: int = None,
+                   page: int = None,

```

Comment:
Instead of setting the page's default value to 1 in resources.py L235, I think you should do that here. That way crud.py will be a little less dependent on `resources.py`. Don't you think?

Reply:
Yes makes sense but the current way avoids an extra check. So if we assign the default value in the function definition of crud.py, I will have to make two statements in resources.py to call get_collection function. 
```
If page == None:
     get_collection(.....) # not providing value for page parameter
else:
     get_collection(...,page=page,....) 
```
Why is it so?
Because if I only use the current statement `get_collection(....,page=page,...)` the `page` value is always being passed(None in the case of missing query parameter), so the default value assignment never happens in the function as a value is always provided.

1) So to set the default value in crud.py one way is to add an extra check as shown in the code snippet above.
2) Manually set the default value in crud.py. 
```
page == None:
     page = 1
```
Yes, it acts differently than setting value in the argument list by ` page: int = 1` because here default value is applied only when page parameter is not passed, if page == None is passed then the default value is not set.

So to set the default value in crud.py I would have to do an extra check(option 1) OR set the default value in the function body of get_collection instead of argument list (option 2).

Reply:
I think option 1 would be a better choice here. I'm saying so because
- It'll make our codebase more explicit i.e if page is None then we call `get_collection` without the page argument and crud.py takes care of the defaults, if we have page value then we pass page value as an argument and the default value is overridden.
- It'll also solve the issue I initially pointed out of making `crud.py` less dependent on `resources.py`

I don't think adding an extra check would hurt us in any way. Thoughts?

Reply:
I agree, I just wanted to show all the options we have so I can get those reviewed and make a better decision. **Going with option 1**